source/grid/grid_tools.cc

```diff
@@ -552,7 +552,7 @@ namespace GridTools
     for (const auto &cell : tria.cell_iterators_on_level(0))
       {
         // Save cell data
-        CellData<dim> cell_data;
+        CellData<dim> cell_data(cell->n_vertices());
```

Comment:
Is this an unrelated change? I don't quite see the connection. Maybe put it into a separate PR?

Reply:
`CellData` takes the `GeometryInfo` equivalent of `n_vertices` as a default argument. This is not correct for simplices and will trigger an assertion later. This change was needed to make `merge_triangulations` work with simplices.

I can't remember which one that was, but I think it was this one
https://github.com/dealii/dealii/blob/5f4d7c1e0dd5afeee39d88bf6fd5d35f16a3c194/source/grid/grid_tools.cc#L594-L606

Reply:
I've used `merge_triangulation` to generate the hybrid meshes for these tests.